### PR TITLE
next batch of PY3 fixes and travis-ci integration

### PR DESCRIPTION
**Pull request by dodobas**
* force order of sets and use assertCountEqual

### CODE REVIEW

File: temba/contacts/tests.py [hunk 1]
```diff
@@ -3498,6 +3498,7 @@ def test_contact_import(self):
         response = self.client.post(customize_url, post_data, follow=True)
         self.assertFormError(response, 'form', None, "Field names can only contain letters, numbers, hypens")
 
+        post_data['column_country_label'] = 'Whatevaar'  # reset invalid label value with a valid one
```
Comment (dodobas) [inline temba/contacts/tests.py]: becuse of different order in PY3, and the fact that `post_data` still had an invalid label, next text was failing

File: temba/contacts/views.py [hunk 2]
```diff
@@ -1301,30 +1301,29 @@ def __init__(self, *args, **kwargs):
...
-                        raise forms.ValidationError(_("Field name '%s' is a reserved word") % label)
-                    used_labels.append(label.lower())
-                else:
-                    # don't allow fields that are dependencies for flows be removed
-                    if field != '__new_field':
-                        from temba.flows.models import Flow
-                        flow = Flow.objects.filter(org=self.org, field_dependencies__in=[field]).first()
-                        if flow:
-                            raise forms.ValidationError(_('The field "%s" cannot be removed while it is still used in the flow "%s"' % (field.label, flow.name)))
+        used_labels = set()
```
Comment (dodobas) [inline temba/contacts/views.py]: key changes, using a set instead of a list, and generator expression in the loop

File: temba/utils/tests.py [hunk 3]
```diff
@@ -1694,15 +1694,24 @@ def assertOrgCounts(qs, counts):
         assertOrgCounts(ContactField.objects.all(), [6, 6, 6])
         assertOrgCounts(ContactGroup.user_groups.all(), [10, 10, 10])
         assertOrgCounts(Contact.objects.filter(is_test=True), [4, 4, 4])  # 1 for each user
-        assertOrgCounts(Contact.objects.filter(is_test=False), [17, 7, 6])
+
+        if six.PY2:
+            # approved by Nic Pottier :)
```
Comment (nicpottier) [inline temba/utils/tests.py]: You mean Rowan!

File: temba/flows/tests.py [hunk 4]
```diff
@@ -4685,7 +4685,7 @@ def test_simple(self):
         self.assertIsNone(run.exit_type)
         self.assertIsNone(run.exited_on)
         self.assertTrue(run.responded)
-        self.assertEqual(run.results, {
+        six.assertCountEqual(self, run.results, {
```
Comment (rowanseymour) [inline temba/flows/tests.py]: don't get why this is needed - dict equality is never based on order, no?

File: temba/msgs/tests.py [hunk 5]
```diff
@@ -2257,7 +2257,7 @@ def assertInDB(self, obj, msg=None):
         fullmsg = "Object %r unexpectedly not found in the database" % obj
         fullmsg += ": " + msg if msg else ""
         try:
-            type(obj).objects.using('direct').get(pk=obj.pk)
+            type(obj).objects.get(pk=obj.pk)
```
Comment (rowanseymour) [inline temba/msgs/tests.py]: as mentioned in slack chat - I think this is being used to test that something is committed thus it is intentionally using a different db connection

File: temba/settings_common.py [hunk 6]
```diff
@@ -978,6 +978,14 @@
         'task': 'squash_contactgroupcounts',
         'schedule': timedelta(seconds=300),
     },
+    "resolve-twitter-ids-task": {
+        'task': 'resolve_twitter_ids_task',
+        'schedule': timedelta(seconds=900)
```
Comment (rowanseymour) [inline temba/settings_common.py]: we completely sure celery is finding these tasks?

Comment (dodobas) [reply temba/settings_common.py]: you are right, I've updated the code, and now the tasks are discovered and executed

I was thrown off by a naming 'clash', as there were two tasks with the same name  `refresh_jiochat_access_tokens` one in channels/tasks and the other one in channels/types/jiochat/tasks ... never checked if twitter was actually running 

I've renamed the one in channels/tasks to `refresh_all_jiochat_access_tokens`

File: temba/msgs/tests.py (hunk 5 above)
Comment (dodobas) [reply temba/msgs/tests.py]: Tried to simply override django settings and add a new configuration, but Django will initializes db connection early in the startup process, so it does not work, I could not find anything similar in django tests, they only define DATABASE configurations in the settings file (https://github.com/django/django/blob/master/tests/test_sqlite.py)

However, simply closing the connection is enough to open a new connection (old connection id='5aaba138.23', new connection id='5aaba149.25'):
```
2018-03-16 10:49:42.774 UTC,"docker","test_temba_test",35,"172.25.0.1:37434",5aaba138.23,3687,"idle",2018-03-16 10:49:28 UTC,,0,LOG,00000,"disconnection: session time: 0:00:13.870 user=docker database=test_temba_test host=172.25.0.1 port=37434",,,,,,,,,""
2018-03-16 10:49:45.201 UTC,,,37,"172.25.0.1:37448",5aaba149.25,1,"",2018-03-16 10:49:45 UTC,,0,LOG,00000,"connection received: host=172.25.0.1 port=37448",,,,,,,,,""                                                                
2018-03-16 10:49:45.202 UTC,"docker","test_temba_test",37,"172.25.0.1:37448",5aaba149.25,2,"authentication",2018-03-16 10:49:45 UTC,2/5291,0,LOG,00000,"connection authorized: user=docker database=test_temba_test",,,,,,,,,"" ```

File: temba/flows/tests.py (hunk 4 above)
Comment (dodobas) [reply temba/flows/tests.py]: true, did not realize that run.results was a dict

File: temba/channels/tests.py [hunk 7]
```diff
@@ -9183,27 +9183,18 @@ def setUp(self):
...
-                    "id": "xy5/5y6O81+/kbWHpLhBoA==",
-                    "name": "ET3",
-                },
-                "message": {
-                    "text": "incoming msg",
-                    "type": "text",
-                    "tracking_data": "3055"
-                }
-            }
+            data = (
```
Comment (rowanseymour) [inline temba/channels/tests.py]: you can also use `json.dumps` with `sort_keys=True` but no biggie

File: temba/msgs/tests.py [hunk 8]
```diff
@@ -2257,7 +2257,9 @@ def assertInDB(self, obj, msg=None):
         fullmsg = "Object %r unexpectedly not found in the database" % obj
         fullmsg += ": " + msg if msg else ""
         try:
-            type(obj).objects.using('direct').get(pk=obj.pk)
+            # close the current connection to the database, so we force it to open a new connection
```
Comment (rowanseymour) [inline temba/msgs/tests.py]: 👍

File: temba/settings_travis.py [hunk 9]
```diff
@@ -8,3 +8,22 @@
 # Flowserver - on Travis we start a GoFlow instance at http://localhost:8800
 # -----------------------------------------------------------------------------------
 FLOW_SERVER_URL = 'http://localhost:8800'
+
+DATABASES = {
```
Comment (rowanseymour) [inline temba/settings_travis.py]: why do we need this?

**Pull request merged**